addons/account/models/account_payment.py

```diff
@@ -875,6 +885,26 @@ def create(self, vals_list):
                     pay.move_id.write(move_vals)
         return payments
 
+    def _get_outstanding_account(self, payment_type):
+        if payment_type not in ('inbound', 'outbound'):
+            return self.env['account.account'].search([], limit=1, order='code')
+
+        account_mapping = {
+            'inbound': ('account_journal_payment_debit_account_id', 'asset_receivable'),
+            'outbound': ('account_journal_payment_credit_account_id', 'liability_current')
+        }
+
+        account_ref, account_type = account_mapping[payment_type]
+        outstanding_account = self.with_context(company_id=self.company_id).env['account.chart.template'].ref(account_ref, raise_if_not_found=False)
+        if not outstanding_account:
+            outstanding_account = self.env['account.account'].search([('type', '=', account_type)], limit=1, order='code')
+
+        if not outstanding_account:
+            # Fallback to any account if no specific account is found
+            outstanding_account = self.env['account.account'].search([], limit=1, order='code')
```

Comment:
Let's just fail, they shouldn't be able to be in any of these cases from the UI with only invoicing installed.